components/WeeklyOverview/index.js

```diff
@@ -58,28 +63,31 @@ export default function WeeklyOverview() {
       ...prevStatus,
       [weekday]: {
         ...prevStatus[weekday],
-        [supplementKey]: false,
+        [supplementKey]: {},
       },
     }));
   };
 
-  const handleCheckSupplement = (weekday, supplementKey) => {
+  const handleCheckSupplement = (weekday, supplementKey, selectedTime) => {
     setSupplementStatus((prevStatus) => ({
       ...prevStatus,
       [weekday]: {
         ...prevStatus[weekday],
-        [supplementKey]: !prevStatus[weekday][supplementKey],
+        [supplementKey]: {
+          ...prevStatus[weekday][supplementKey],
+          [selectedTime]: !prevStatus[weekday]?.[supplementKey]?.[selectedTime] || false,
+        },
       },
     }));
   };
```

Comment:
I find this part of your code rather difficult to read, so I would suggest refactoring it for more readability:

```suggestion
const handleCheckSupplement = (weekday, supplementKey, selectedTime) => {
  setSupplementStatus(prevStatus => {
    // Extract current day's status or an empty object if not defined.
    const currentDayStatus = prevStatus[weekday] || {};

    // Extract current supplement's status or an empty object if not defined.
    const currentSupplementStatus = currentDayStatus[supplementKey] || {};

    // Toggle the current time's status.
    const newTimeStatus = !currentSupplementStatus[selectedTime] || false;

    // Construct the updated status.
    const updatedStatus = {
      ...prevStatus,
      [weekday]: {
        ...currentDayStatus,
        [supplementKey]: {
          ...currentSupplementStatus,
          [selectedTime]: newTimeStatus
        }
      }
    };

    return updatedStatus;
  });
};
```

This should work just as fine, while being a lot easier to understand.
You can of course remove the comments :)